src/Tizen.Uix.Sticker/Tizen.Uix.Sticker/StickerProvider.cs

```diff
@@ -0,0 +1,301 @@
+        /// <exception cref="InvalidOperationException">This exception can be due to operation failed.</exception>
+        public static Task InsertData(string jsonPath)
+        {
+            var task = new TaskCompletionSource<bool>();
+            _insertDelegate = (ErrorCode errorCode, IntPtr userData) =>
+            {
+                if (errorCode != ErrorCode.None)
+                {
+                    Log.Error(LogTag, "Exception occurred while inserting");
+                    throw ExceptionFactory.CreateException(errorCode);
+                }
+                else
+                {
+                    task.SetResult(true);
+                }
+            };
+            ErrorCode error = StickerProviderInsertDataByJsonFile(_handle, jsonPath, _insertDelegate, IntPtr.Zero);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "InsertData Failed with error " + error);
+                task.SetException(ExceptionFactory.CreateException(error));
+            }
+            return task.Task;
+        }
+
+        /// <summary>
+        /// Updates a sticker data in the sticker database.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <feature>http://tizen.org/feature/ui_service.sticker</feature>
+        /// <param name="data">The sticker data to be updated.</param>
+        /// <exception cref="NotSupportedException">The required feature is not supported.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">
+        /// This can occur due to the following reasons:
+        /// 1) This exception can be due to operation failed.
+        /// 2) This exception can be due to file exists.
+        /// 3) This exception can be due to no such file.
+        /// </exception>
+        public static void UpdateData(StickerData data)
+        {
+            ErrorCode error = StickerProviderUpdateData(_handle, data.SafeStickerDataHandle);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "UpdateData Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// Deletes a sticker data in the sticker database.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <feature>http://tizen.org/feature/ui_service.sticker</feature>
+        /// <param name="data">The sticker data to be deleted.</param>
+        /// <exception cref="NotSupportedException">The required feature is not supported.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">This exception can be due to operation failed.</exception>
+        public static void DeleteData(StickerData data)
```

Comment:
it's always interesting to ask hat happens if user tries removing non-existing SickerData

Reply:
ArgumentException occurs when developers delete non-existing data.